tfx/experimental/templates/taxi/README.md

```diff
@@ -24,3 +24,63 @@ about the dataset in [Google BigQuery](https://cloud.google.com/bigquery/).
+
+This tutorial serves as an extension to `template.ipynb` tutorial, thus you will also use [Taxi Trips dataset](
+https://data.cityofchicago.org/Transportation/Taxi-Trips/wrvz-psew)
+released by the City of Chicago. We strongly encourage you to try modifying the components prior to utilizing stub executors.
+
+### Step 1. Record the pipeline outputs in Google Cloud Storage
+
+Since this tutorial assumes that you have completed `template.ipynb` up to step 6, a successful pipeline run must have been saved in the MLMD, which can be accessed using gRPC server. 
+
+Open a Terminal and run the following commands:
+
+> `$ gcloud container clusters get-credentials <cluster_name> --zone <compute_zone> --project {GOOGLE_CLOUD_PROJECT}`
+
+> `$ nohup kubectl port-forward deployment/metadata-grpc-deployment -n <namespace> {PORT}:8080 &`
+
+Open another terminal and run the following command:
+> `$
+python tfx/experimental/pipeline_testing/pipeline_recorder.py \
+--output_dir=$output_dir \
+--host=$host \
+--port=$port \
+--pipeline_name=$pipeline_name
+`
+
+`--output_dir` should be set to a path in Google Cloud Storage (e.g. `gs://<project_name>-kubeflowpipelines-default/testdata`). 
+`--host` and `--port` are hostname and port of the metadata grpc server to connect to MLMD. You may choose any unused port and 'localhost' for hostname.
+
+In `template.ipynb` tutorial, the pipeline name is set as "my_pipeline" by default, so set `pipeline_name="my_pipeline"`. If you have modified the pipeline name when running the template tutorial, you should modify the `--pipeline_name` accordingly.
+
+### Step 2. Enable Stub Executors in Kubeflow DAG Runner
+>**Double-click to open `kubeflow_dag_runner.py`**. 
+Uncomment `supported_launcher_classes` argument of `KubeflowDagRunnerConfig` to be able to launch stub executors (Tip: search for comments containing `TODO(step 11):`),  Make sure to save `kubeflow_dag_runner.py` after you edit it.
+
+
+
+>**Double-click to change directory to `launcher` and double-click again to open `stub_component_launcher.py`.**
+Make sure to set `test_data_dir` to the GCS directory where KFP outputs are recorded, or `output_dir` (Tip: search for comments containing `TODO(StubExecutor):`). Customize the list `self.stubbed_component_ids`, or ids of components that should be replaced with BaseStubExecutor(Tip: search for comments containing `TODO(StubExecutor):`). You may additionally insert custom stub executor in `self.stubbed_component_map` with component id as a key and custom stub executor class as value (Tip: search for comments containing `TODO(StubExecutor):`). Make sure to save `stub_component_launcher.py` after you edit it.
```

Comment:
Would you mind adding some descriptions for users who didn't start from the template? For example, we might add snippets or links to the corresponding files or code segments.